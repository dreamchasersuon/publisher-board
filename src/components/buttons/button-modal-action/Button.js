import React from 'react';
import PropTypes from 'prop-types';
import './Buttons.css';

export default function Button({ onPress, label, testId }) {
  return (
    <button
      data-testid={testId}
      onClick={onPress}
      type="button"
      className="Modal-action"
    >
      {label}
    </button>
  );
}

Button.propTypes = {
  onPress: PropTypes.func,
  label: PropTypes.string,
  testId: PropTypes.string,
};
