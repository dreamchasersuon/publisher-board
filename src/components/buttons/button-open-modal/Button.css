.Header-button {
  border: 1px solid #e5e5e5;
  border-radius: 4px;
  color: #666;
  background-color: transparent;
  margin-left: auto;
  outline: none;
  cursor: pointer;
  transition: all 0.6s cubic-bezier(0.39, 0.575, 0.565, 1);
}

.Header-button:hover {
  border-color: black;
  color: black;
}

.Popover-button {
  padding: 10px 20px;
  font-size: 14px;
  color: #444;
  text-align: left;
  border: none;
  outline: none;
  border-bottom: 1px solid #e5e5e5;
  cursor: pointer;
  background-color: white;
  transition: all 0.6s cubic-bezier(0.39, 0.575, 0.565, 1);
}

.Popover-button:last-child {
  border: none;
}

.Popover-button:hover {
  background-color: #fafafa;
}
