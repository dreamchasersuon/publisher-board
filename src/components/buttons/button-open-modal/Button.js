import React from 'react';
import PropTypes from 'prop-types';
import './Button.css';

export default function Button({
  title,
  onPress,
  dataName,
  className,
  testId,
}) {
  return (
    <button
      data-testid={testId}
      data-name={dataName}
      onMouseDown={onPress}
      className={className}
    >
      {title}
    </button>
  );
}

Button.propTypes = {
  title: PropTypes.string,
  onPress: PropTypes.func,
  dataName: PropTypes.string,
  className: PropTypes.string,
  testId: PropTypes.string,
};
