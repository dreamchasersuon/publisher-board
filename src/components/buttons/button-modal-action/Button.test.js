import React from 'react';
import renderer from 'react-test-renderer';
import Button from './Button';

const handleOnPress = jest.fn();

test('renders modal action button with given label and action', () => {
  const tree = renderer
    .create(<Button label="close" onPress={handleOnPress} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
