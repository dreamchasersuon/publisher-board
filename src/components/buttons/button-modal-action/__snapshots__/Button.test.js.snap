// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders modal action button with given label and action 1`] = `
<button
  className="Modal-action"
  onClick={[MockFunction]}
  type="button"
>
  close
</button>
`;
